eq-author/src/App/page/Design/answers/DateRange/index.test.js

```diff
@@ -2,36 +2,80 @@ import React from "react";
 import { shallow } from "enzyme";
 import DateRange from "./";
 import createMockStore from "tests/utils/createMockStore";
-
-const answer = {
-  id: "1",
-  label: "Lorem ipsum",
-  childAnswers: [
-    { id: "1from", label: "Period from" },
-    { id: "1to", label: "Period to" },
-  ],
-};
+import { DATE_LABEL_REQUIRED } from "constants/validationMessages";
+import { render as rtlRender } from "tests/utils/rtl";
 
 describe("DateRange", () => {
-  let handleChange;
-  let handleUpdate;
-  let store;
+  let handleChange, handleUpdate, store, props;
 
   beforeEach(() => {
     handleChange = jest.fn();
     handleUpdate = jest.fn();
     store = createMockStore();
+
+    props = {
+      answer: {
+        id: "1",
+        label: "",
+        type: "DateRange",
+        description: "test",
+        guidance: "",
+        secondaryLabel: "",
+        properties: {required: false},
+        displayName: "Untitled answer",
+        qCode: "yuky",
+        validationErrorInfo:
+        {
+          errors: [],
+        },
+      },
+    };
   });
 
   it("should render", () => {
     const wrapper = shallow(
       <DateRange
         onChange={handleChange}
         onUpdate={handleUpdate}
-        answer={answer}
+        answer={props.answer}
         store={store}
       />
     );
     expect(wrapper).toMatchSnapshot();
   });
+
+  it("should render an error message when date label input is empty", () => {
+    const error = [
+      {
+        errorCode: "ERR_VALID_REQUIRED",
+        field: "label",
+        id: "71b5f8a2-5de1-4a9a-8923-acd7fd3b7080",
+        type: "answer",
+        __typename: "ValidationError",
+      },
+    ];
+    props.answer.validationErrorInfo.errors = error;
+
+    const { getAllByText } = rtlRender(<DateRange onUpdate={handleUpdate} {...props} />);
+
+    expect(getAllByText(DATE_LABEL_REQUIRED)).toBeTruthy();
```

Comment:
Hi Shane, using `getAllByText` indeed asserts whether the error message is there, but it will return any and all instances of that error message regardless of where it is applied on the page. Therefore, I'm not sure we can't be sure that the error you're expecting is coming from the first label and not the second.

There's a singular varient of `getAllByText()` called `getByText()` you could use. Here's a link to the documentation: 

https://testing-library.com/docs/bs-react-testing-library/examples/#getbytext

https://testing-library.com/docs/dom-testing-library/cheatsheet/#queries

Reply:
Or you could combine the tests and use getAllByText and check it has two errors present. One for the label and one for the secondary label

Reply:
> Or you could combine the tests and use getAllByText and check it has two errors present. One for the label and one for the secondary label

My only issue with this is that it wouldn't assert that you can have one without the other.

Reply:
Well spotted, thanks Matt